### DISCUSSION TITLE
frame start and frame end field option

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#155] [General]
Hi, presently in iw3 there is a field for video start and end time that is great for 2d to 3d encoding for mp4. Is it possible to add a new field option in iw3 for frame start and frame end number. This would be a useful option for those chosing the beginning and end frame they want to use when using export disparity rather than converting the whole video. Also this would be a clever work around for someone trying to resume from the last frame number generated before their pc crashed during an export disparity but instead of using a resume button per say, the user just looks in their folder at the last export the was just generated and they go in iw3 and just type in the frame number to start from of the last frame that didnt generate to start from, great work around for resume

**Comment by nagadomi** [comment]
iw3 cannot handle the correct frame numbers of the original video because it performs FPS conversion when decoding the video.
It will change when FPS setting is changed.
Also, if Export has a resume feature, it should not be necessary I think.

**Reply by kaelsonofkrypto** [reply]
sad to hear its not doable, it would have been a really beneficial feature in IW3 if it had the option. The main reason I suggested it is that, in IW3, if you have a section with an artifact, you could easily use another program to find the exact frame number where the issue occurred and just edit that section. The current seconds feature is great for finding a section to test, but it's harder to be precise because different programs handle timing differently. However, every program is consistent with frame numbers. Knowing the exact frame number allows users to precisely edit the frames where an issue happens. This would be a great workaround for fixing small errors.

Having both the current time structure and the frame number option would be valuable. The existing time structure is useful for selecting a five-minute clip, while the frame start and end feature would be more beneficial for those needing to cut sections precisely. This way, they would know the exact frame numbers to replace the issue section with the corrected frames.

**Reply by nagadomi** [reply]
You can split the video with a different program before processing it with iw3.
That should be more compatible with a different program.
If you use a lossless video format for intermediate video, there should be no degradation.

**Reply by kaelsonofkrypto** [reply]
Hi there, on export disparity, I'm experiencing an issue when resuming my 3D conversion project after an outage. Sometimes, the frames get messed up, requiring me to reconvert a section of the affected frames.
While I can theoretically use another software to convert the 2d video of the specific frame range, starting from where the issue occurred, and then re-import the 2d video of just the errored section back into iw3 to reconvert the depth map again just on that clip, this creates a new problem. When I re-import the video clip back into iw3, iw3 will rename the frames starting from 00000000, which disrupts the frame sequence. If its only a few frames, its fine to rename, however this issue becomes particularly challenging when working with hundreds or even thousands of frames that had errors. And renaming them all back to what it was originally is impractical. I understand that adding a start and end frame function may be difficult due to the reliance on FPS, but I'm wondering This is why having a frame start and frame end field for the naming is something that would really help if it was in iw3. If this is something that really can't be added in iw3, is there any alternative solution that can address this problem and help preserve the frame sequence naming convention without the need for renaming. Thank you for considering my request.

**Reply by nagadomi** [reply]
Resume feature starts processing from a specific frame number (pts). It starts from the last frame number in the rgb/depth directory minus the batch size.

If the last few frames are corrupted.
1. Delete the corrupt and subsequent files, then resume

If you want to repair 00003000-00005000,
1. Move files after 00005000 from both rgb/depth to another folder (backup)
2. Delete files after 00003000 from both rgb/depth
3. Resume and cancel when progress exceeds 5000
4. Restore(move files) after 00005000 from backup

**Reply by kaelsonofkrypto** [reply]
Thanks, i will try this
